src/types.d.ts

```diff
@@ -23,6 +23,11 @@ export interface GrafanaQuery extends DataQuery {
 
 export interface GenericOptions extends DataSourceJsonData {}
 
+export interface VariableQuery {
+  query: string;
+  format: string;
```

Comment:
can we enumerate options through string union?

```suggestion
  format: 'string' | 'json';
```

also I wonder whether to use `'string'` or some better name 🤔

Reply:
> can we enumerate options through string union?

Oh cool, I didn't even know we could do that. Is there something else we should update in our checks for string vs json?

> also I wonder whether to use `'string'` or some better name 🤔

Yeah I think we talked about "raw" but after looking at some of the other datasources I feel like the concept of "raw" implies that it is the base representation of any format type. For example, "raw json" can be conflated with "raw query" when really they sort of imply different things, neither of which is the fact that the type of entry people are using in this case is in fact a string. Therefore, I went with "string" because that's what it is, a plain old query **string**.

I don't feel strongly about that though, it's whatever you want to to... I'm just happy we're going to be able to supply the template variables as JSON. 🥇